service-base/src/main/java/org/eclipse/hono/service/registration/AbstractRegistrationService.java

```diff
@@ -271,6 +320,37 @@ protected final JsonObject getAssertionPayload(final String tenantId, final Stri
         return result;
     }
 
+    /**
+     * Creates the payload of the assert Registration response message.
+     * <p>
+     * The returned JSON object may contain the {@link RegistrationConstants#FIELD_VIA} property of the device's
+     * registration information and may also contain <em>default</em> values registered for the device under key
+     * {@link RegistrationConstants#FIELD_PAYLOAD_DEFAULTS}.
+     *
+     * @param tenantId The tenant the device belongs to.
+     * @param deviceId The device to create the assertion token for.
+     * @param registrationInfo The device registration information.
+     * @param resultHandler The handler to invoke with the result of this operation.
+     */
+    protected final void getAssertionPayload(final String tenantId, final String deviceId,
```

Comment:
It would be preferable to use a `Future<JsonObject>` return value here, instead of using the `Handler` param.
We just had a community call where we decided that we only declare the documented APIs on the Hono website as public (mailinglist announcement to follow). In that sense we can remove the old `getAssertionPayload` method above and use the `Future` return value here.

Reply:
The reason for choosing the approach with a `Handler` parameter was that this was done similar in other protected or public methods of this class such as `getDevice` or `resolveGroupMember`. Thus my intention was not to mix the approaches. But considering your description of the decision from the community call, I will change it here to  `Future<JsonObject>' which I prefer as well.